frontend/src/pages/modelServing/screens/projects/NIMServiceModal/NIMModelListSection.tsx

```diff
@@ -52,21 +55,29 @@ const NIMModelListSection: React.FC<NIMModelListSectionProps> = ({
     getModelNames();
   }, [dashboardNamespace]);
 
-  const getSupportedModelFormatsInfo = (name: string) => {
+  const getSupportedModelFormatsInfo = (key: string) => {
+    const lastHyphenIndex = key.lastIndexOf('-');
+    const name = key.slice(0, lastHyphenIndex);
+    const version = key.slice(lastHyphenIndex + 1);
+
     const modelInfo = modelList.find((model) => model.name === name);
     if (modelInfo) {
       return {
         name: modelInfo.name,
-        version: modelInfo.latestTag,
+        version,
       };
     }
     return null;
   };
 
-  const getNIMImageName = (name: string) => {
+  const getNIMImageName = (key: string) => {
+    const lastHyphenIndex = key.lastIndexOf('-');
+    const name = key.slice(0, lastHyphenIndex);
+    const version = key.slice(lastHyphenIndex + 1);
+
     const imageInfo = modelList.find((model) => model.name === name);
     if (imageInfo) {
-      return `nvcr.io/${imageInfo.namespace}/${name}:${imageInfo.latestTag}`;
+      return `nvcr.io/${imageInfo.namespace}/${name}:${version}`;
     }
     return '';
   };
```

Comment:
This is logic that feels very brittle just reading it. I'll need to fire this up and look at the data flowing through here -- I can't imagine slicing at specific intervals to the most ideal way to do this.